airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -58,7 +58,7 @@ def state(self, value: MutableMapping[str, Any]) -> None:
 
     @property
     def primary_key(self) -> PrimaryKeyType:
-        return self.config.primary_key
+        return self.config.primary_key or self.get_parser().get_parser_defined_primary_key(self.config)
```

Comment:
Nice. Very simple/elegant, I think. 👍 

The only thing I worry about is that if this gets called multiple times (as would be encouraged by the property-type declaration), I'm not sure if there's be a performance hit, since both of the get_...() methods could be doing real work.

Not something we have to change, but just wanted to call it out. One option would be to cache, either manually or with an decorator (@cached, @lrucache, etc.). Some of these decorator do or don't work with dictionaries, but I don't think a dictionary is possible here. 

Realistically, though, despite them being methods I think both of those get_() operations are simple returns, so I doubt this would _actually_ be an issue.

Wdyt?